chain/store/snapshot.go

```diff
@@ -442,10 +445,19 @@ func (s *walkScheduler) processTask(t walkTask, workerN int) error {
 		b: blk,
 	}
 
+	// We exported the ipld block. If it wasn't a CBOR block, there's nothing
+	// else to do and we can bail out early as it won't have any links
+	// etc.
+	if t.c.Prefix().Codec != cid.DagCBOR || t.c.Prefix().MhType == mh.IDENTITY {
```

Comment:
While it likely doesn't matter (at the moment, we _should_ scan for links in "identity" blocks. In theory, I can inline dag-cbor into an identity hash.

Reply:
@Stebalien I never knew the reasons to skip identity hashes, it was something inherited from @frrist code. I can't say if there's an instance of an identity hash in the whole chain history (? genesis or something?)

Reply:
Hm. So, we shouldn't include blocks for identity hashes, but we should technically be traversing them and including their children, if any.

On the other hand, I can guarantee that this won't make a difference in practice (at the moment, at least). So we should be good to merge this as-os.

Reply:
ok, happy to send follow-up in a few days.

Reply:
fwiw I inherited it from: https://github.com/filecoin-project/lotus/blob/e945c0d6f25f490323ec542d535f584022a20a36/chain/store/snapshot.go#L668